### PR TITLE
Data Augumentation -- Rotation

### LINKED ISSUE #57: Data Augumentation -- Rotation
This is a sub-issue for #55

This issue needs to be resolved asap.

Need to create a method for Data_Augumentation.py file for rotating the image.

### PR DESCRIPTION
**Pull request by rubyruins**
# Description:

Added a method in Data_Augumentation.py file for rotating the image based on the angle given by the user. Image is rotated by the specific angle in degrees in the counterclockwise direction The images are added to class/RotatedImages. Exception handling is added in case of invalid angle values (not int). 

It is used in the following way:

```
flow = Flow( 'dataset_dir' , 'data_dir' )
operations = {'rotate': 100}
flow.applyAugmentation( operations )
```

Fixes #57 

## Type of change:

- [x] New feature (non-breaking change which adds functionality)
 
# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.

# Screenshots / Video:

Original, Rotation by 45, 90, 540
<p float="left">
  
   
  
  
</p>

### CODE REVIEW

**Comment by chebroluharika** [comment]
@rubyruins please remove tests file from this PR.

**Comment by rubyruins** [comment]
@chebroluharika, sorry I had forgotten to remove it. I have removed it now.

**Comment by chebroluharika** [comment]
@rubyruins please add necessary documentation in readme inside flow2ml folder

File: flow2ml/Data_Augumentation.py [hunk 1]
```diff
@@ -50,3 +51,31 @@ def applyFlip(self,classPath):
...
+    try:
+      os.mkdir(classPath+"/RotatedImages")    
+    except:
+      raise Exception("Unable to create directory for rotated images.")
+
+    for image in list(os.listdir(classPath)):
+      # Read image
+      img = cv2.imread(classPath+"/"+image)
+      if not isinstance(self.operations['rotate'], int):
+        raise Exception("Rotation angle must be an integer.")
```
Comment (chebroluharika) [inline flow2ml/Data_Augumentation.py:70]: I just want to understand the reason behind throwing except if angle is not in integer datatype. Why can’t we take floor/ceil of number and apply rotation.

Comment (rubyruins) [reply flow2ml/Data_Augumentation.py:70]: I had actually put this for checking if it was any other type except for int (like string). But yes, we can round it off instead if it is a decimal. I'll make the changes by rounding it off to the nearest decimal, if that's alright.

**Comment by rubyruins** [comment]
@chebroluharika, I've made the changes

**Review by chebroluharika — APPROVED** [review]

**Pull request merged**